Sources/Foundation/DateFormatter.swift

```diff
@@ -64,8 +64,15 @@ open class DateFormatter : Formatter {
         }
 
         // range.length is updated with the last position of the input string that was parsed
-        guard range.length == string.length else {
-            // The whole string was not parsed
+        guard let swiftRange = Range(NSRange(range), in: string) else {
+            preconditionFailure("Incorrect range \(range) in \(string)")
```

Comment:
If this is triggered is it due to a bug? should it be `fatalError()`  instead?

Reply:
Oh, right. Should be `fatalError()`, indeed.

Reply:
Done. And rebased on latest master.